src/devices/auriol_afw2a1.c

```diff
@@ -0,0 +1,178 @@
+/** @file
+    Auriol AFW 2 A1 sensor.
+
+    Copyright (C) 2019 LiberationFrequency
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+/*
+version=0.01.001 beta
+
+Lidl Auriol AFW 2 A1 sensor.
+IAN 311588
+
+Technical data for the external sensor:
+Temperature measuring range/accuracy:       -20 to +65°C (-4 to +149°F) / ±1.5 °C (± 2.7 °F)
+Relative humidity measuring range/accuracy: 20 to 99% / ± 5%
+Relative humidity resolution:               1%
+Transmission frequencies:                   433 MHz (ch1:~433919300,ch2:~433915200,ch3:~433918000, various?)
+Transmission output:                        < 10 dBm / < 10 mW
+
+The ID is retained even if the batteries are changed.
+The device has three channels and a transmit button.
+
+Data layout:
+The sensor transmits 12 identical messages of 36 bits in a single package each ~60 seconds, depending on the temperature.
+e.g.:
+[00] {36} 90 80 ba a3 a0 : 10010000 10000000 10111010 10100011 1010
+...
+[11] {36} 90 80 ba a3 a0 : ...
+     0           1           2           3           4
+ 9    0      8    0      b    a      a    3      a    0
+|1001|0000| |1000|0000| |1011|1010| |1010|0011| |1010|
+|id       | |chan|temp            | |fix |hum        |
+--------------------------------------------------------
+10010000  = id=0x90=144; 8 bit
+1000      = channel=0x8=8=8-8+1=1; 4 bit
+            0=ch1 & bat_low (ch + 1)
+            1=ch2 & bat_low (ch + 1)
+            2=ch3 & bat_low (ch + 1)
+            4=ch1 & tx_button & bat_low (ch - 4 + 1)
+            5=ch2 & tx_button & bat_low (ch - 4 + 1)
+            6=ch3 & tx_button & bat_low (ch - 4 + 1)
+            8=ch1 & bat_ok (ch - 8 + 1)
+            9=ch2 & bat_ok (ch - 8 + 1)
+            a=ch3 & bat_ok (ch - 8 + 1)
+            c=ch1 & tx_button & bat_ok (ch - 12 + 1)
+            d=ch2 & tx_button & bat_ok (ch - 12 + 1)
+            e=ch3 & tx_button & bat_ok (ch - 12 + 1)
+0000      = temperature leading sign,
+            1110=0xe(-51.1°C to -25.7°C),
+            1111=0xf(-25.6°C to - 0.1°C),
+            0000=0x0(  0.0°C to  25,5°C),
+            0001=0x1( 25.6°C to  51.1°C),
+            0010=0x2( 51.2°C to  76.7°C); 4 bit
+10111010  = temperature=0xba=186=18,6°C; 8 bit
+1010      = fixed; 4 bit 
+0011 1010 = humidity=0x3a=58%; 8 bit
+*/
+
+#include "decoder.h"
+
+static int auriol_afw2a1_decode(r_device *decoder, bitbuffer_t *bitbuffer) {
+    data_t *data;
+    int row;
+    uint8_t *b;
+    int id;
+    int channel;
+    int battery_low = 1;
+    int tx_button = 1;
+    int16_t temp_raw;
+    float temp_c;
+    int humidity;
+
+    row = bitbuffer_find_repeated_row(bitbuffer, 12, 36);
+    if (bitbuffer->num_rows != 12)
+        return DECODE_ABORT_EARLY;
```

Comment:
Do you really need this? It would easily break if the reception isn't prefect.

Reply:
No, we can leave that out, if you say so. It should prevent that unnecessarily wrong sensors are read out.

Reply:
Usually if you expect to see e.g. 12 rows we search for 7 identical ones (more than half).
But since there is no checksum you can be a little more restrictive (exactly 12 x 36 seems to fragile though)